### PR TITLE
Get data for donated books

### PR DESCRIPTION
**Pull request by ghost**
## Description
Get id, name, author, avg_rating, number of borrowed
## Reference Requirements
- Product Backlog ID: PBI-10
- Task ID: 10.1

## Migrations
YES

## Impacted Areas in Application
List general components of the application that this PR will affect:
   None

## SQL query and `explain`
- Select id, name, author, avg_rating of books donated by user have id = 1
  select books.id, books.name, books.author, books.avg_rating,
 (select count(*) from `borrowings` where `books`.`id` = `borrowings`.`book_id` and 
 `borrowings`.`deleted_at` is null) as `number_of_borrowed` 
 from `books`
 inner join donators on donators.id = books.donator_id
 inner join users on donators.user_id = users.id
 where users.id = 1

- Refer to [Using Explain with MySQL]
image

### CODE REVIEW

**Comment by at-luanle** [comment]
LGMT

File: app/Http/Controllers/Admin/BookController.php [hunk 1]
```diff
@@ -17,21 +18,36 @@ class BookController extends Controller
     public function create()
     {
         $fields = [
-            'id',
-            'name'
+        'id',
```
Comment (at-huydq) [inline app/Http/Controllers/Admin/BookController.php]: sao xoá 4 space rồi ?

File: app/Http/Controllers/Admin/BookController.php [hunk 2]
```diff
@@ -17,21 +18,36 @@ class BookController extends Controller
...
+     * @param Request $request filter if have
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
-        $books  = Book::with('borrowings')->withCount('borrowings')->sortable()->paginate(config('define.page_length'));
+      
+        $uid = $request->uid;
```
Comment (at-huydq) [inline app/Http/Controllers/Admin/BookController.php]: check has rồi mới lấy dữ liệu ra chứ ? 
move line 38, 39 xuống dưới `if ($request->has('uid') && $request->has('filter'))`

File: app/Http/Controllers/Admin/BookController.php [hunk 3]
```diff
@@ -17,21 +18,36 @@ class BookController extends Controller
...
+      
+        if ($request->has('uid') && $request->has('filter')) {
+            if ($filter == "donated") {
+                $books = Book::whereHas('donator', function ($query) use ($uid) {
+                    $query->where('user_id', '=', $uid);
+                })->withCount('donator')->sortable()->paginate(config('define.page_length'));
+            }
+        } else {
+            $books  = Book::with('borrowings')->withCount('borrowings')->sortable()->paginate(config('define.page_length'));
+        };
```
Comment (at-huydq) [inline app/Http/Controllers/Admin/BookController.php]: sao lại có `;` ở đây ?

**Review by ghost — COMMENTED** [review]
check my comment and fix conficts

File: app/Http/Controllers/Admin/BookController.php [hunk 4]
```diff
@@ -53,6 +54,20 @@ public function index(Request $request)
         $books = $books->withCount('borrowings')
             ->sortable()
             ->paginate(config('define.page_length'));
+      
+        if ($request->has('uid') && $request->has('filter')) {
+            $uid = $request->uid;
+            $filter = $request->filter;
+
+            if ($filter == "donated") {
```
Comment (at-huydq) [inline app/Http/Controllers/Admin/BookController.php]: đặt const trong define rồi call nhé, ko dc dùng hardcode.

File: resources/views/backend/users/index.blade.php [hunk 5]
```diff
@@ -18,32 +18,33 @@
...
+            </thead>
               <tbody>
                 @foreach ($users as $user)
                 <tr>
                   <td>{{ $user->id }}</td>
                   <td>{{ $user->employee_code }}</td>
                   <td><a href="{{ route('users.show', ['employeeCode' => $user->employee_code])}}">{{ $user->name }} </a></td>
                   <td>{{ $user->email }}</td>
-                  <td>{{ $user->total_donated }}</td>
+                  <td><a href="{{ route('books.index',['uid' => $user->id, 'filter' => 'donated']) }}">{{ $user->total_donated }}</td>
```
Comment (at-huydq) [inline resources/views/backend/users/index.blade.php]: như trên, ko dc dùng hardcode

**Review by ghost — COMMENTED** [review]
please check my comment

File: resources/views/backend/users/index.blade.php (hunk 5 above)
Comment (at-nghiavo) [reply resources/views/backend/users/index.blade.php]: dạ, em đã fix

**Comment by at-huydq** [comment]
LGTM.

**Pull request merged**